SeniorSlideshowProject/src/Slides/AudioState.java

```diff
@@ -1,31 +1,58 @@
 package Slides;
 
+
+import java.io.File;
+import java.io.IOException;
 import java.io.Serializable;
 
+import javax.sound.sampled.AudioInputStream;
+import javax.sound.sampled.AudioSystem;
+import javax.sound.sampled.UnsupportedAudioFileException;
+
 public class AudioState implements Serializable {
 	
 	private static final long serialVersionUID = 1L;
+	private String fileName;
+	private AudioInputStream audioFile;
+	private long audioTime;
 	
-	private String filename;
-	private double audioTime;
-	
-	public AudioState(String name)
+	public AudioState(AudioInputStream inputStream, String name)
```

Comment:
You can remove the String from the constructor and get rid of the fileName member variable. We just want the inputStream that you will serialize.